cmd/compute-domain-controller/resourceclaimtemplate.go

```diff
@@ -228,11 +241,14 @@ func (m *BaseResourceClaimTemplateManager) RemoveFinalizer(ctx context.Context,
 		return fmt.Errorf("error updating ResourceClaimTemplate: %w", err)
 	}
 
+	// Update mutation cache after successful update
+	m.mutationCache.Mutation(newRCT)
+
 	return nil
 }
 
 func (m *BaseResourceClaimTemplateManager) AssertRemoved(ctx context.Context, cdUID string) error {
-	rcts, err := getByComputeDomainUID[*resourceapi.ResourceClaimTemplate](ctx, m.informer, cdUID)
+	rcts, err := getByComputeDomainUID[*resourceapi.ResourceClaimTemplate](ctx, m.informer.GetIndexer(), cdUID)
```

Comment:
Interesting. Which of the following describes the goal of this method better?

1. "assert removed": confirming that an action was performed (confirm that _I_ explicitly removed this just before"; a check that could be done with just process-local state)
2. "assert not exists", in global state (the API server)

Reply:
2 -- in which case it will only be gone from the informer cache once it is also gone from the API server

Reply:
I see. So, in case of informer lag being significant, we're just spinning on this `AssertRemoved()` until the deletion has propagated back to the local informer state. Sounds good.